packages/react-icons/src/utils/useIconState.tsx

```diff
@@ -22,7 +27,7 @@ export const useIconState = <TBaseAttributes extends (React.SVGAttributes<SVGEle
   
     const styles = useRootStyles();
   
-    state.className = mergeClasses(styles.root, state.className);
+    state.className = mergeClasses(styles.root, dir === 'rtl' && styles.rtl, state.className);
```

Comment:
We need a way to only apply this mirroring behavior to icons that need it. One option is a prop like `autoMirror` that defaults to false. We could then change the default to true for specific icons from some list (like rtl.txt) when generating the React components for the icons.